### PR TITLE
feature: countless pagination

### PR DESCRIPTION
**Pull request by Paul-Bob**
Fixes [#1748](https://github.com/avo-hq/avo/issues/1748)

### CODE REVIEW

File: docs/3.0/resources.md [hunk 1]
```diff
@@ -691,3 +692,14 @@ class Avo::Resources::Project < Avo::BaseResource
   self.index_query = -> { query.unscoped }
 end
 ```
+:::
+
+:::option self.countless
```
Comment (adrianthedev) [inline docs/3.0/resources.md:697]: This doesn't tell anything about how it will look like. WE need to say that the page numbers will not be visible, etc.

**Review by adrianthedev — CHANGES_REQUESTED** [review]
Let's talk a little bit about how that pagination will look like with that setting.
Also let's attach screenshots for each type of pagination.

And let's change the `self.countless` API.

**Comment by Paul-Bob** [comment]
@adrianthedev the docs were improved

File: docs/2.0/resources.md [hunk 2]
```diff
@@ -567,3 +567,81 @@ end
...
+# Or as hash:
+self.pagination = {
+  type: :default,
+  size: [1, 2, 2, 1],
+}
+```
+
+The exposed pagination setting above have the default value for each key.
+
+### `type`<br><br>
```
Comment (adrianthedev) [inline docs/2.0/resources.md:597]: Can these be displayed as `:::option`? Maybe they will stand out better?

Comment (adrianthedev) [reply docs/2.0/resources.md:597]: CleanShot 2023-11-17 at 17 03 55@2x

**Review by adrianthedev — APPROVED** [review]

File: docs/2.0/resources.md (hunk 2 above)
Comment (Paul-Bob) [reply docs/2.0/resources.md:597]: @adrianthedev i tried it but that will make the option visible on the right lateral sidebar and it looks like "type" and "size" becomes an resource option. Would be nice to have "nested options" so we can declare :::option inside an :::option

Comment (adrianthedev) [reply docs/2.0/resources.md:597]: ok. leave it like that.
Shouldn't we add the docs to 3.0 too?

**Pull request merged**